src/awscr-s3/client.cr

```diff
@@ -296,5 +296,21 @@ module Awscr::S3
     private def http
       @http
     end
+
+    private def encode(object)
+      String.build do |io|
+        URI.encode(object, io, space_to_plus: true) do |byte|
+          # check for characters we want to encode
+          # otherwise fall back to default implementation
+          # https://github.com/crystal-lang/crystal/blob/5999ae29beacf4cfd54e232ca83c1a46b79f26a5/src/uri/encoding.cr#L70-L72
+          !should_encode?(byte) && (URI.reserved?(byte) || URI.unreserved?(byte))
+        end
+      end
+    end
+
+    private def should_encode?(byte)
+      char = byte.unsafe_chr
+      char == '='
```

Comment:
`=` is a "reserved" character which is why it isn't being encoded

https://github.com/crystal-lang/crystal/blob/5999ae29beacf4cfd54e232ca83c1a46b79f26a5/src/uri/encoding.cr#L166

Reply:
`should_encode?` should also return true for apostrophe. Fails on Amazon AWS for any object key with an apostrphe in it without that.